eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java

```diff
@@ -0,0 +1,245 @@
+import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpVariantsWrapper;
+import uk.ac.ebi.eva.accession.dbsnp.test.MongoTestConfiguration;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@TestPropertySource("classpath:test-variants-writer.properties")
+@ContextConfiguration(classes = {MongoConfiguration.class, MongoTestConfiguration.class,
+        SubmittedVariantAccessioningConfiguration.class})
+public class DbsnpVariantsWriterTest {
+
+    private static final int TAXONOMY_1 = 3880;
+
+    private static final int TAXONOMY_2 = 3882;
+
+    private static final long EXPECTED_ACCESSION = 10000000000L;
+
+    private static final int START_1 = 100;
+
+    private static final Long CLUSTERED_VARIANT = 12L;
+
+    private static final VariantType VARIANT_TYPE = VariantType.SNV;
+
+    private static final Long SUBMITTED_VARIANT = 15L;
+
+    private static final Boolean SUPPORTED_BY_EVIDENCE = true;
+
+    private static final Boolean MATCHES_ASSEMBLY = false;
+
+    private static final Boolean ALLELES_MATCH = true;
+
+    private static final Boolean VALIDATED = false;
+
+    private DbsnpVariantsWriter dbsnpVariantsWriter;
+
+    private Function<ISubmittedVariant, String> hashingFunctionSubmitted;
+
+    private Function<IClusteredVariant, String> hashingFunctionClustered;
+
+    private Function<ISubmittedVariant, String> hashingFunction;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Before
+    public void setUp() throws Exception {
+        dbsnpVariantsWriter = new DbsnpVariantsWriter(mongoTemplate);
+        hashingFunctionSubmitted = new DbsnpSubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        hashingFunctionClustered = new DbsnpClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        mongoTemplate.dropCollection(DbsnpSubmittedVariantEntity.class);
+        mongoTemplate.dropCollection(DbsnpClusteredVariantEntity.class);
+        mongoTemplate.dropCollection(DbsnpSubmittedVariantOperationEntity.class);
+        hashingFunction = new DbsnpSubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+    }
+
+    @Test
+    public void writeBasicVariant() throws Exception {
+        DbsnpVariantsWrapper wrapper = buildSimpleWrapper();
+
+        dbsnpVariantsWriter.write(Collections.singletonList(wrapper));
+
+        List<DbsnpSubmittedVariantEntity> ssEntities = mongoTemplate.find(new Query(),
+                                                                          DbsnpSubmittedVariantEntity.class);
+        assertEquals(1, ssEntities.size());
+        assertEquals(wrapper.getSubmittedVariants().get(0), ssEntities.get(0));
+
+        assertRsWasStored(wrapper);
```

Comment:
The assert style is inconsistent: submitted variants are checked in the test method, but the clustered variant is checked delegating into another. If needed created another method that receives the number of expected SS, and that will allow you to loop through the retrieved entities in a generic way.